packages/core/src/browser/view-container.ts

```diff
@@ -0,0 +1,935 @@
+                    dragImage.innerText = this.wrapped.title.label;
+                    document.body.appendChild(dragImage);
+                    dataTransfer.setDragImage(dragImage, -10, -10);
+                    setTimeout(() => document.body.removeChild(dragImage), 0);
+                }
+            }, false),
+            addEventListener(this.node, 'dragend', () => this.canBeDropTarget = true, false),
+            addEventListener(this.node, 'dragover', style, false),
+            addEventListener(this.node, 'dragleave', unstyle, false),
+            addEventListener(this.node, 'drop', event => {
+                const { dataTransfer } = event;
+                if (dataTransfer) {
+                    const moveId = dataTransfer.getData('view-container-dnd');
+                    if (moveId && moveId !== this.id) {
+                        this.moveBeforeEmitter.fire(moveId);
+                    }
+                    unstyle(event);
+                }
+            }, false)
+        );
+    }
+
+    protected createContent(): { header: HTMLElement, body: HTMLElement, disposable: Disposable } {
+        const disposable = new DisposableCollection();
+        const { header, disposable: headerDisposable } = this.createHeader();
+        const body = document.createElement('div');
+        body.classList.add('body');
+        this.node.appendChild(header);
+        this.node.appendChild(body);
+        disposable.push(headerDisposable);
+        return {
+            header,
+            body,
+            disposable,
+        };
+    }
+
+    protected createHeader(): { header: HTMLElement, disposable: Disposable } {
+        const disposable = new DisposableCollection();
+        const header = document.createElement('div');
+        header.classList.add('theia-header', 'header');
+        disposable.push(addEventListener(header, 'click', () => {
+            this.collapsed = !this.collapsed;
+        }));
+
+        const toggleIcon = document.createElement('span');
+        toggleIcon.classList.add(EXPANSION_TOGGLE_CLASS);
+        if (this.collapsed) {
+            toggleIcon.classList.add(COLLAPSED_CLASS);
+        }
+        header.appendChild(toggleIcon);
+
+        const title = document.createElement('span');
+        title.classList.add('label', 'noselect');
+        title.innerText = this.wrapped.title.label;
+        header.appendChild(title);
+
+        if (ViewContainerPart.ContainedWidget.is(this.wrapped)) {
+            for (const { tooltip, execute, className } of this.wrapped.toolbarElements.filter(e => e.enabled !== false)) {
+                const toolbarItem = document.createElement('span');
+                toolbarItem.classList.add('element');
+                if (typeof className === 'string') {
+                    toolbarItem.classList.add(...className.split(' '));
+                } else if (Array.isArray(className)) {
+                    toolbarItem.classList.add(...className);
```

Comment:
This is unsafe; the array of CSS classes can be `['this is incorrect', 'correct']`.